src/test/scala/org/broadinstitute/dsde/vault/services/mock/MockServers.scala

```diff
@@ -0,0 +1,144 @@
+    mockDMServer.stop()
+    mockBossServer.stop()
+  }
+
+  def startUpDM(): Unit = {
+    val ubam = new UBam(
+      "testing_id",
+      Map(("bam", "http://localhost:8080/path/to/bam"), ("bai", "http://localhost:8080/path/to/bai")),
+      Map("testAttr" -> "testValue")
+    )
+
+    val ubamResponse = new UBamIngestResponse(
+      "testing_id",
+      Map(("bam", "http://localhost:8080/path/to/bam"), ("bai", "http://localhost:8080/path/to/bai"))
+    )
+
+    val lookupResponse = new EntitySearchResult(
+      "testing_id",
+    "ubam"
+    )
+
+    mockDMServer = startClientAndServer(8989)
+    mockDMServer
+      .when(
+        request()
+          .withMethod("GET")
+          .withPath("/ubams/testing_id")
+      ).respond(
+        org.mockserver.model.HttpResponse.response()
+          .withHeaders(header)
+          .withBody(ubam.toJson.prettyPrint)
+          .withStatusCode(200)
+      )
+
+    mockDMServer
+      .when(
+        request()
+          .withMethod("GET")
+          .withPath("/ubams/12345-67890-12345")
+      ).respond(
+        org.mockserver.model.HttpResponse.response()
+          .withHeaders(header)
+          .withStatusCode(404)
+      )
+
+    mockDMServer.when(
+      request()
+        .withMethod("POST")
+        .withPath("/ubams")
+    ).respond(
+        org.mockserver.model.HttpResponse.response()
+          .withHeaders(header)
+          .withBody(ubam.toJson.prettyPrint)
+          .withStatusCode(200)
+      )
+
+    mockDMServer.when(
+      request()
+        .withMethod("PUT")
+        .withPath("/ubams")
+    ).respond(
+        org.mockserver.model.HttpResponse.response()
+          .withBody("HTTP method not allowed, supported methods: GET")
+          .withStatusCode(405)
+      )
+
+    mockDMServer.when(
+      request()
+        .withMethod("GET")
+        .withPath("/query/ubam/uniqueTest/.*")
+    ).respond(
+        org.mockserver.model.HttpResponse.response()
+          .withHeader(header)
+          .withBody(lookupResponse.toJson.prettyPrint)
+          .withStatusCode(200)
+      )
+
+  }
+
+  def startUpBoss(): Unit = {
+
+    val list = List("one","two")
+    val bossObject = new BossCreationObject(
+      "bossSample", "localhost", 24, "localOwner", list, list, Some("testing_id"), Some("34"), Some(true)
+    )
+
+    val bossResponse = new BossResolutionResponse(
+      10, Option("application/json"), Option("md5string"), "http://localhost:8988/objects/testing_id/resolve"
+    )
+
+    mockBossServer = startClientAndServer(8988)
+
+    mockBossServer.when(
+      request()
+        .withMethod("POST")
+        .withPath("/objects/")
+    ).respond(
+        org.mockserver.model.HttpResponse.response()
+          .withHeaders(header)
+          .withBody(bossObject.toJson.prettyPrint)
+          .withStatusCode(200)
+      )
+
+    mockBossServer.when(
+      request()
+        .withMethod("POST")
+        .withPath("/objects/testing_id/resolve")
```

Comment:
Curious, were we ever able to figure out accessing regular expressions in the path? Or something to the effect of PathParams?